grakn-graql-shell/src/main/java/ai/grakn/graql/GraqlShell.java

```diff
@@ -502,75 +487,31 @@ private void printLicense() {
         result.append("\n");
         scanner.close();
 
-        this.print(result.toString());
+        this.println(result.toString());
     }
 
-    private boolean executeQuery(String queryString) throws IOException {
-        // Split query into chunks
-        Iterable<String> splitQuery = Splitter.fixedLength(QUERY_CHUNK_SIZE).split(queryString);
-
-        for (String queryChunk : splitQuery) {
-            Json jsonObject = Json.object(
-                    ACTION, ACTION_QUERY,
-                    QUERY, queryChunk
-            );
-            session.sendJson(jsonObject);
-        }
-
-        session.sendJson(Json.object(ACTION, ACTION_END));
-        handleMessagesFromServer();
+    private void executeQuery(String queryString) throws IOException {
+        handleGraknExceptions(() -> {
+            Stream<Query<?>> queries = tx.graql().parser().parseList(queryString);
+            queries.flatMap(query -> query.results(printer())).forEach(this::println);
+        });
 
         // Flush the console so the output is all displayed before the next command
         console.flush();
-
-        return true;
-    }
-
-    private void handleMessagesFromServer() {
-        session.getMessagesUntilEnd().forEach(this::handleMessage);
-    }
-
-    /**
-     * Handle the given server message
-     *
-     * @param message the message to handle
-     */
-    private void handleMessage(Json message) {
-        switch (message.at(ACTION).asString()) {
-            case ACTION_QUERY:
-                String result = message.at(QUERY_RESULT).asString();
-                print(result);
-                break;
-            case ACTION_TYPES:
-                Set<String> types = message.at(TYPES).asJsonList().stream().map(Json::asString).collect(toSet());
-                graqlCompleter.setTypes(types);
-                break;
-            case ACTION_ERROR:
-                System.err.print(message.at(ERROR).asString());
-                errorOccurred = true;
-                break;
-            case ACTION_PING:
-                // Ignore
-                break;
-            default:
-                throw new RuntimeException("Unrecognized message: " + message);
-        }
     }
 
     private void setDisplayOptions(Set<String> displayOptions) throws IOException {
-        session.sendJson(Json.object(
-                ACTION, ACTION_DISPLAY,
-                DISPLAY, displayOptions
-        ));
+        displayAttributes = displayOptions.stream().map(tx::getAttributeType).collect(toImmutableSet());
```

Comment:
Curiosity: what does this method do? The name is maybe a bit vague, but I am sure it's just because I don't have a context, so I am just curious.

Reply:
Update: it's not OK because we use that URI to check if the server is running or not, which must be over HTTP.

I don't know what to do about that :(

Reply:
It lets you display attributes on the results:
```
>>> display name
>>> match $x isa person;
$x isa person has name "Alice";
$x isa person has name "Bob";
...
```

Very handy! No one uses it.

Reply:
🥇 👍 💯